src/Slices/Dashboard/UI/Charts/LineChart.tsx

```diff
@@ -0,0 +1,244 @@
+        minDomain={{ y: isStacked || min === 0 ? 0 : min * 0.9 }}
+        name={`chart-${title}`}
+        padding={{
+          bottom: isStacked ? 85 : 60, // Adjusted to accommodate legend
+          left: 60,
+          right: 25,
+          top: 20,
+        }}
+        themeColor={ChartThemeColor.multiUnordered}
+        height={300}
+        width={width}
+      >
+        <CustomAxis
+          tickLabelComponent={
+            <StyledChartLabel
+              dx={({ index }) => (index == timestamps.length - 1 ? -20 : 0)}
+              lineHeight={1.4}
+            />
+          }
+          tickFormat={(x) => {
+            const date = new Date(x).toUTCString();
+            return date.slice(5, 16) + "\n" + date.slice(17, 25);
+          }}
+          style={{
+            ticks: { size: 10 },
+          }}
+        />
+        <ChartAxis
+          dependentAxis
+          showGrid
+          label={label}
+          style={{
+            grid: { stroke: "#E5E4E2" },
+          }}
+        />
+        {isStacked ? (
+          <ChartStack>
+            {metrics.reverse().map(({ name, data }, index) => (
+              <ChartArea
+                data={data.map((value, index) => {
+                  return { x: timestamps[index], y: value };
+                })}
+                name={name}
+                key={name + index}
+                style={{
+                  data: {
+                    stroke:
+                      colorTheme[name] === undefined
+                        ? colorTheme.default
+                        : colorTheme[name],
+                    fill:
+                      colorTheme[name] === undefined
+                        ? colorTheme.default
+                        : colorTheme[name],
+                  },
+                }}
+              />
+            ))}
+          </ChartStack>
+        ) : (
+          <ChartGroup>
+            {metrics.map(({ name, data }, index) => (
+              <ChartLine
+                data={data.map((value, index) => {
+                  return {
+                    x: timestamps[index],
+                    y:
+                      value % 1 === 0 ? value : Math.round(value * 1000) / 1000,
+                  };
+                })}
+                name={name}
+                key={name + index}
+                style={{
+                  data: {
+                    stroke:
+                      colorTheme[name] === undefined
+                        ? colorTheme.default
+                        : colorTheme[name],
+                  },
+                }}
+              />
+            ))}
+          </ChartGroup>
+        )}
+        <ChartGroup>
+          {metrics.map(({ name, data }, metricIndex) => (
+            <ChartScatter
+              data={data.map((value, index) => {
+                return {
+                  x: timestamps[index],
+                  y:
+                    value === null
+                      ? null
```

Comment:
Dunno if it's the same double ternary as in the other PR, but same comment anyway. This is not clean ;)